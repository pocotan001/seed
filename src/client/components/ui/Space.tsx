import styled from "styled-components";
import margin, { MarginProps } from "../styles/extends/margin";

type SpaceProps = MarginProps;

const Space = styled("div")<SpaceProps>`
  ${margin};
`;

export default Space;
