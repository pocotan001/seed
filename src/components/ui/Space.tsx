import styled from "styled-components";
import withMargin, {
  MarginProps
} from "~/components/styles/extends/withMargin";

interface SpaceProps extends MarginProps {
  children: React.ReactNode;
}

const Space = styled<SpaceProps, "div">("div")`
  ${withMargin};
`;

export default Space;
