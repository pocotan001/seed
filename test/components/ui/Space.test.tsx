import * as React from "react";
import * as renderer from "react-test-renderer";
import Space from "~/components/ui/Space";

describe("<Space>", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Space>alo</Space>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
