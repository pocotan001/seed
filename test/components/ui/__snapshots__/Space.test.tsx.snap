// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space> should render correctly 1`] = `
<div
  className="Space-sc-9fq1rg-0 kOMwYl"
>
  alo
</div>
`;
